samples/amber.cc

```diff
@@ -178,42 +186,101 @@ int main(int argc, const char** argv) {
     return 0;
   }
 
-  if (options.input_filename.empty()) {
+  if (options.input_filenames.empty()) {
     std::cerr << "Input file must be provided." << std::endl;
     return 2;
   }
 
-  auto data = ReadFile(options.input_filename);
-  if (data.empty())
-    return 1;
-
-  amber::Amber am;
-  amber::Recipe recipe;
-  amber::Result result = am.Parse(data, &recipe);
-  if (!result.IsSuccess()) {
-    std::cerr << result.Error() << std::endl;
-    return 1;
-  }
-
-  if (options.parse_only)
-    return 0;
-
+  amber::Result result;
+  sample::ConfigHelper config_helper;
   amber::Options amber_options;
   amber_options.engine = options.engine;
-  result = am.Execute(&recipe, amber_options);
-  if (!result.IsSuccess()) {
-    std::cerr << result.Error() << std::endl;
-    return 1;
+  if (options.input_filenames.size() > 1UL && !options.parse_only) {
+    amber_options.config = amber::MakeUnique<amber::VulkanEngineConfig>();
+
+    amber::VulkanEngineConfig* config =
+        static_cast<amber::VulkanEngineConfig*>(amber_options.config.get());
+    std::tie(result, *config) = config_helper.CreateVulkanConfig();
+    if (!result.IsSuccess()) {
+      std::cerr << result.Error() << std::endl;
+      return 1;
+    }
   }
 
-  if (!options.buffer_filename.empty()) {
-    // TODO(dsinclair): Write buffer file
-    assert(false);
+  std::vector<amber::Amber> ambers(options.input_filenames.size());
```

Comment:
This isn't needed, you don't need to store the Amber object, just create a new one later